registrar/apps/enrollments/models.py

```diff
@@ -0,0 +1,126 @@
+        through='OrganizationProgramMembership'
+    )
+
+
+class OrganizationProgramMembership(TimeStampedModel):
+    """
+    Table that captures the relationship between Programs and Organizations.
+    """
+    class Meta(object):
+        app_label = 'enrollments'
+        unique_together = [
+            ('program', 'organization',)
+        ]
+
+    program = models.ForeignKey(Program, on_delete=models.CASCADE)
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    history = HistoricalRecords()
+
+
+class LearnerStatus(TimeStampedModel):
+    """
+    Represents various statuses that a learner can have
+    e.g. "SYNCED_WITH_LMS", "PENDING", etc.
+    """
+    key = models.CharField(max_length=64, db_index=True, null=False)
+
+
+class Learner(TimeStampedModel):
+    """
+    Model that represents an LMS user who may be enrolled in programs and courses.
+    """
+    lms_id = models.IntegerField(db_index=True, null=True)
+
+    # for storing id field provided by org
+    external_id = models.CharField(max_length=255, db_index=True, null=True)
+
+    email = models.EmailField(db_index=True)
+    status = models.ForeignKey(LearnerStatus, on_delete=models.PROTECT)
+    history = HistoricalRecords()
+
+
+class ProgramEnrollmentStatus(TimeStampedModel):
+    """
+    Model that represents possible Program enrollment statuses.
+    """
+    key = models.CharField(max_length=64, db_index=True, null=False)
+
+
+class LearnerProgramEnrollment(TimeStampedModel):
+    """
+    Model that represents the relationship between a Learner and a Program.
+    """
+    # we shouldn't be able to delete any of these fields out from under
+    # ourselves if an enrollment record is using them.
+    learner = models.ForeignKey(Learner, on_delete=models.PROTECT)
+    program = models.ForeignKey(Program, on_delete=models.PROTECT)
+    status = models.ForeignKey(ProgramEnrollmentStatus, on_delete=models.PROTECT)
+    history = HistoricalRecords()
+
+
+class CourseRun(TimeStampedModel):
+    """
+    Model that represents a course-discovery CourseRun entity.
+    """
+    catalog_uuid = models.UUIDField(db_index=True, null=True)
+    title = models.CharField(max_length=512)
+    key = models.CharField(max_length=255, db_index=True)
+    organizations = models.ManyToManyField(
+        Organization,
+        related_name='course_runs',
+        through='OrganizationCourseRunMembership'
+    )
+
+
+class OrganizationCourseRunMembership(TimeStampedModel):
+    """
+    Model that represents the relationship between CourseRuns and Organizations.
+    """
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    course_run = models.ForeignKey(CourseRun, on_delete=models.CASCADE)
+
+
+class CourseRunEnrollmentStatus(TimeStampedModel):
+    """
+    Model that represents possible CourseRun enrollment statuses.
+    """
+    key = models.CharField(max_length=64, db_index=True, null=False)
```

Comment:
How would we leverage the status here to drive other logic within code if these are defined during run time? For example, we want to update the LearnerCourseRunEnrollment later from "Pending" to "active". These are values defined by systems, not by users, right?

Reply:
I was thinking of hard-cording some enums in the code, and they would also be persisted.  Kind of like we do for course modes.

Reply:
I see, you just haven't write down what those ENUMs are. OK. I got ya